File: server/py_server/models/rag_document.py
from sqlalchemy import Column, DateTime, Integer, String, Text
from sqlalchemy.sql import func

from config.base import Base


class RAGDocument(Base):
    __tablename__ = "rag_documents"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, nullable=False, index=True)
    document_name = Column(String(255), nullable=False)
    chunk_index = Column(Integer, nullable=False, default=0)
    snippet = Column(Text, nullable=True)
    metadata_json = Column(Text, nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

File: server/js_server/models/RAGDocument.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/db');

const RAGDocument = sequelize.define('RAGDocument', {
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true,
  },
  user_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
  },
  document_name: {
    type: DataTypes.STRING(255),
    allowNull: false,
  },
  snippet: {
    type: DataTypes.TEXT,
    allowNull: true,
  },
  metadata_json: {
    type: DataTypes.TEXT,
    allowNull: true,
  },
}, {
  tableName: 'rag_documents',
  timestamps: true,
  createdAt: 'created_at',
  updatedAt: false,
});

module.exports = RAGDocument;
